frodo.js

```diff
@@ -3,45 +3,43 @@
  *
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.